pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -275,7 +276,7 @@ stdenv.mkDerivation ({
 
     echo setupCompileFlags: $setupCompileFlags
     ${nativeGhcCommand} $setupCompileFlags --make -o Setup -odir $TMPDIR -hidir $TMPDIR $i
-
+    ${optionalString splitCheck "${nativeGhcCommand} $setupCompileFlags ${./ListTests.hs} -o ListTests"}
```

Comment:
the newline here could be restored, before the `runHook`, i had omited it to avoid a ghc rebuild during initial testing